src/main/java/org/generationcp/middleware/dao/ProjectDAO.java

```diff
@@ -1,180 +1,256 @@
-			final Criteria criteria = this.getSession().createCriteria(Project.class);
-			for (Map.Entry<ProgramFilters, Object> entry : filters.entrySet()) {
-				final ProgramFilters filter = entry.getKey();
-				final Object value = entry.getValue();
-				criteria.add(Restrictions.eq(filter.getStatement(), value));
-			}
-
-			final int start = pageSize * (pageNumber - 1);
-			final int numOfRows = pageSize;
-
-			criteria.setFirstResult(start);
-			criteria.setMaxResults(numOfRows);
-			return criteria.list();
-		} catch (HibernateException e) {
-			throw new MiddlewareException(
-				"Error in getProjectsByFilter(start=" + pageNumber + ", numOfRows=" + pageSize + "): " + e.getMessage(), e);
-		}
-	}
-
-	public long countProjectsByFilter(final Map<ProgramFilters, Object> filters) throws MiddlewareException {
-		try {
-			final Criteria criteria = this.getSession().createCriteria(Project.class);
-			for (Map.Entry<ProgramFilters, Object> entry : filters.entrySet()) {
-				final ProgramFilters filter = entry.getKey();
-				final Object value = entry.getValue();
-				criteria.add(Restrictions.eq(filter.getStatement(), value));
-			}
-
-			return criteria.list().size();
-		} catch (HibernateException e) {
-			throw new MiddlewareException("Error in countProjectsByFilter(): " + e.getMessage(), e);
-		}
-	}
-}
+/*******************************************************************************
+ * Copyright (c) 2012, All Rights Reserved.
+ *
+ * Generation Challenge Programme (GCP)
+ *
+ *
+ * This software is licensed for use under the terms of the GNU General Public License (http://bit.ly/8Ztv8M) and the provisions of Part F
+ * of the Generation Challenge Programme Amended Consortium Agreement (http://bit.ly/KQX1nL)
+ *
+ *******************************************************************************/
+
+package org.generationcp.middleware.dao;
+
+import org.generationcp.middleware.domain.workbench.RoleType;
+import org.generationcp.middleware.exceptions.MiddlewareException;
+import org.generationcp.middleware.exceptions.MiddlewareQueryException;
+import org.generationcp.middleware.pojos.workbench.CropType;
+import org.generationcp.middleware.pojos.workbench.Project;
+import org.generationcp.middleware.pojos.workbench.WorkbenchUser;
+import org.generationcp.middleware.service.api.program.ProgramFilters;
+import org.hibernate.Criteria;
+import org.hibernate.HibernateException;
+import org.hibernate.SQLQuery;
+import org.hibernate.criterion.Restrictions;
+import org.hibernate.transform.AliasToEntityMapResultTransformer;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * DAO class for {@link Project}.
+ *
+ */
+public class ProjectDAO extends GenericDAO<Project, Long> {
+
+	public static final String GET_PROJECTS_BY_USER_ID =
+		"SELECT  "
+			+ "    p.* "
+			+ "	FROM "
+			+ "    	workbench_project p "
+			+ "        	INNER JOIN "
+			+ "    	workbench_crop wc ON p.crop_type = wc.crop_name "
```

Comment:
You can save a join to `workbench_crop `table here by joining `workbench_project.crop_type` to `crop_persons.crop_name`